File: Decorator/Java/PizzaMenu/FreshTomato.java
package Decorator.PizzaMenu;

public class FreshTomato extends ToppingDecorator {

    Pizza pizza;

    public FreshTomato(Pizza pizza) {
        this.pizza = pizza;
    }

    @Override
    public String getName() {
        return "FreshTomato " + pizza.getName();
    }

    @Override
    public int getCost() {
        return pizza.getCost() + 20;
    }
}

File: Decorator/Cpp/PizzaMenu/src/FreshTomato.cpp
/*
 * FreshTomato.cpp
 *
 *  Created on: Jan 18, 2021
 *      Author: nquang
 */

#include "FreshTomato.h"

FreshTomato::FreshTomato(Pizza *component) {
	this->component_ = component;
}
int FreshTomato::getCost() const {
	return this->component_->getCost() + this->cost;
}
string FreshTomato::getName() const {
	return this->name + this->component_->getName();
}
string FreshTomato::getDescription() const {
	return this->component_->getDescription() + this->name;
}
